internal/manifest/manifest_test.go

```diff
@@ -0,0 +1,166 @@
+package manifest_test
+
+import (
+	"os"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/klauspost/compress/zstd"
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/chisel/internal/manifest"
+)
+
+type manifestContents struct {
+	Paths    []manifest.Path
+	Packages []manifest.Package
+	Slices   []manifest.Slice
+}
+
+var manifestTests = []struct {
+	summary string
+	input   string
+	mfest   manifestContents
+	error   string
+}{
+	{
+		summary: "All types",
+		input: `
+{"jsonwall":"1.0","schema":"1.0","count":13}
```

Comment:
It will look better if we indent the content for readability of the struct, and then have a function to dedent it inside the test itself.

Reply:
Done in https://github.com/canonical/chisel/pull/144/commits/8600db66f7bbf65d126a310fca11205c76797a5f.